drivers/interrupt_controller/intc_plic.c

```diff
@@ -29,13 +29,39 @@
 #define PLIC_IRQS        (CONFIG_NUM_IRQS - CONFIG_2ND_LVL_ISR_TBL_OFFSET)
 #define PLIC_EN_SIZE     ((PLIC_IRQS >> 5) + 1)
 
+#define EDGE_TRIG_TYPE   (PLIC_MAX_PRIO + 0x1080)
+#define EDGE_TRIG_SHIFT  5
+
 struct plic_regs_t {
 	uint32_t threshold_prio;
 	uint32_t claim_complete;
 };
 
 static int save_irq;
 
+/**
+ * @brief return edge irq value or zero
+ *
+ * In the event edge irq is enable this will return the trigger
+ * value of the irq. In the event edge irq is not supported this
+ * routine will return 0
+ *
+ * @param irq IRQ number to add to the trigger
+ *
+ * @return irq value when enabled 0 otherwise
+ */
+static int riscv_plic_is_edge_irq(uint32_t irq)
+{
+#if defined(CONFIG_PLIC_SUPPORTS_EDGE_IRQ)
```

Comment:
Could this be a `if (IS_ENABLED(..))` or even `if (!IS_ENABLED(..))`?

Reply:
Do you mean avoid the use of preprocessor magic?